File: Dot examples/ASP Book Examples/Book Examples/Chapter15/VB/App_Code/ProductConstructor.vb

Public Class ProductConstructor

    Public Sub New(ByVal name As String)
        Me.New(name, 0, String.Empty)
    End Sub

    Public Sub New(ByVal name As String, ByVal price As Decimal)
        Me.New(name, price, String.Empty)
    End Sub

    Public Sub New(ByVal name As String, ByVal price As Decimal, ByVal description As String)
        ' Use name, price, and description
    End Sub

End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter15/CS/App_Code/ProductConstructor.cs
using System;

public class ProductConstructor
{
    public ProductConstructor(string name)
        : this(name, 0, String.Empty) { }

    public ProductConstructor(string name, decimal price)
        : this(name, price, String.Empty) { }

    public ProductConstructor(string name, decimal price, string description)
    {
        // Use name, price, and description
    }
}

